### PR TITLE
Shiping labels: Address editing unit tests

### PR DESCRIPTION
**Pull request by 0nko**
Adds unit tests for address editing.

**To test:**

Run `EditShippingLabelAddressViewModelTest`.

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/EditShippingLabelAddressViewModelTest.kt [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+    @Before
+    fun setup() {
+        whenever(dataStore.getCountries()).thenReturn(countries)
+        whenever(dataStore.getStates("US")).thenReturn(states)
+        whenever(dataStore.getStates("SK")).thenReturn(emptyList())
+        whenever(resourceProvider.getString(any())).thenAnswer { i -> i.arguments[0].toString() }
+
+        createViewModel()
+
+        clearInvocations(viewModel, savedState, addressValidator, resourceProvider, dataStore, site)
```
Comment (hichamboushaba) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/EditShippingLabelAddressViewModelTest.kt]: NP, do we need this method here? AFAIK JUnit will create a new instance of the test class for each test, which means the invocations get reset each time, and Mockito's javadoc discourages from using clearInvocations unless necessary.

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/EditShippingLabelAddressViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+    }
+
+    private fun createViewModel() {
+        viewModel = spy(
```
Comment (hichamboushaba) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/EditShippingLabelAddressViewModelTest.kt]: NP, do we need to spy on the instance here?

**Review by hichamboushaba — DISMISSED** [review]
Nice work @0nko, the tests are working well, I left two comments, they are not blocking, just for discussing their usefulness.

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/EditShippingLabelAddressViewModelTest.kt (hunk 2 above)
Comment (0nko) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/EditShippingLabelAddressViewModelTest.kt]: Nope :)

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/EditShippingLabelAddressViewModelTest.kt (hunk 1 above)
Comment (0nko) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/EditShippingLabelAddressViewModelTest.kt]: Hmm. I can't remember why exactly I started doing this but something was failing for one of the unit tests I wrote and this fixed it. But these tests are fine without it, so I'll get rid of it.

**Comment by 0nko** [comment]
Thanks @hichamboushaba! Ready for another round.

**Comment by hichamboushaba** [comment]
Looks good @0nko, but the lint check is failing, please check it.

**Comment by 0nko** [comment]
It's been fixed.

**Review by hichamboushaba — APPROVED** [review]
:shipit:

**Pull request merged**